Mathlib/Analysis/CStarAlgebra/CStarVec.lean

```diff
@@ -0,0 +1,329 @@
+
+@[simp] theorem add_apply [Add A] {i : n} : (v + w) i = v i + w i := rfl
+
+@[simp] theorem sub_apply [Sub A] {i : n} : (v - w) i = v i - w i := rfl
+
+@[simp] theorem smul_apply (c : R) [SMul R A] {i : n} : (c • v) i = c • v i := rfl
+
+@[simp] theorem neg_apply [Neg A] {i : n} : (-v) i = -v i := rfl
+
+theorem finset_sum_fn {ι : Type*} {s : Finset ι} [AddCommMonoid A] {f : ι → CStarVec n A} :
+    ∑ i ∈ s, ofFun (f i) = (fun j => ∑ i ∈ s, f i j) := by
+  rw [Finset.sum_fn]
+  rfl
+
+end basic
+
+section inner
+
+variable {n R A : Type*} [Fintype n] [NonUnitalNormedRing A] [StarRing A] [NormedSpace ℂ A]
+  [PartialOrder A] [StarOrderedRing A] [CStarRing A] [IsScalarTower ℂ A A] [SMulCommClass ℂ A A]
+  [StarModule ℂ A]
+
+instance instInner : Inner A (CStarVec n A) where
+  inner v w := ∑ i : n, star (v i) * w i
+
+noncomputable instance instNorm : Norm (CStarVec n A) where
+  norm v := √‖⟪v, v⟫_A‖
+
+lemma inner_eq_sum {v w : CStarVec n A} : ⟪v, w⟫_A = ∑ i : n, star (v i) * w i := rfl
+
+lemma norm_eq_sum {v : CStarVec n A} : ‖v‖ = √‖∑ i : n, star (v i) * v i‖ := rfl
+
+lemma inner_single_eq_entry [DecidableEq n] {v : CStarVec n A} {i : n} {a : A} :
+    ⟪ofFun (Pi.single i a), v⟫_A = star a * v i := by
+  simp [inner_eq_sum, ofFun_apply, Pi.single_apply]
+  have hmain :
+      (fun x => star (if x = i then a else 0) * v x)
+        = fun x => if x = i then star a * v x else 0 := by
+    ext
+    split_ifs with h <;> simp [h, mul_assoc]
+  simp [hmain, Finset.sum_ite]
+
+@[simp] lemma norm_single {i : n} {a : A} [DecidableEq n] : ‖ofFun (Pi.single i a)‖ = ‖a‖ := by
+  simp [norm_eq_sum, Pi.single_apply, CStarRing.norm_star_mul_self]
+
+instance instCStarModule : CStarModule A (CStarVec n A) where
+  inner_add_right {v} {w₁} {w₂} := by
+    simp only [inner_eq_sum, add_apply, mul_add, Finset.sum_add_distrib]
+  inner_self_nonneg {x} := by
+    simp only [inner]
+    exact Finset.sum_nonneg fun i _ => star_mul_self_nonneg (x i)
+  inner_self {x} := by
+    refine ⟨fun h => ?_, ?_⟩
+    · simp only [inner] at h
+      have : ∀ i ∈ Finset.univ, 0 ≤ star (x i) * x i := fun i _ => star_mul_self_nonneg (x i)
+      rw [Finset.sum_eq_zero_iff_of_nonneg this] at h
+      ext i
+      simp only [zero_apply]
+      exact (CStarRing.star_mul_self_eq_zero_iff (x i)).mp <| h i (Finset.mem_univ i)
+    · rintro rfl; simp only [inner, CStarVec.zero_apply, star_zero, mul_zero, Finset.sum_const_zero]
+  inner_op_smul_right {a} {x} {y} := by
+    simp only [inner, smul_apply, smul_eq_mul, mul_assoc]
+    apply Eq.symm
+    simp [Finset.sum_mul, mul_assoc]
+  inner_smul_right_complex {z} {x} {y} := by
+    simp only [inner_eq_sum, smul_apply, smul_mul_assoc]
+    apply Eq.symm
+    simp only [Finset.smul_sum, mul_smul_comm]
+  star_inner x y := by simp only [inner, star_sum, star_mul, star_star]
+  norm_eq_sqrt_norm_inner_self v := rfl
+
+lemma norm_entry_le_norm [DecidableEq n] [CompleteSpace A] {v : CStarVec n A} {i : n} :
```

Comment:
Is this "entry" terminology used in other parts of the library? 
```suggestion
lemma norm_apply_le_norm [DecidableEq n] [CompleteSpace A] {v : CStarVec n A} {i : n} :
```

Reply:
It's also used a bit in `Matrix`, but it arguably makes more sense there. But yeah, `apply` is a bit more consistent with the rest of mathlib here.